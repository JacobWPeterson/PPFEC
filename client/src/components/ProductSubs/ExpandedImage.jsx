import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  align-items: center;
  justify-content: center;
  z-index: 5;
  max-width: 85vw;
  max-height: 90vh;
  margin: auto;
  overflow: hidden;
  box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
`;

const Expanded = styled.img`
  ${(props) => props.zoomed && 'transform: scale(2.5)'};
  height: 100%;
  width: 100%;
  cursor: ${(props) => (props.zoomed ? 'zoom-out' : 'zoom-in')};
  // background-position-x: ${(props) => props.x};
  // background-position-y: ${(props) => props.y};
`;

class ExpandedImage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      zoomed: false,
      x: 0,
      y: 0,
    };
    this.expanded = React.createRef();
    this.imageClickHandler = this.imageClickHandler.bind(this);
    this.trackMouse = this.trackMouse.bind(this);
  }

  trackMouse() {
    const node = this.expanded.current;

    node.addEventListener('mousemove', (e) => {
      console.log('hello');
      this.setState({
        x: -e.offsetX,
        y: -e.offsetY,
      });
      // node.style.setProperty('--x', `${-e.offsetX}px`);
      // node.style.setProperty('--y', `${-e.offsetY}px`);
    });
  }

  imageClickHandler(event) {
    event.stopPropagation();
    this.setState((prevState) => ({ zoomed: !prevState.zoomed }), this.trackMouse);
  }

  render() {
    return ReactDOM.createPortal(
      <Wrapper>
        <Expanded
          key={this.props.id}
          src={this.props.src}
          alt={this.props.alt}
          zoomed={this.state.zoomed}
          ref={this.expanded}
          x={this.state.x}
          y={this.state.y}
          onMouseDown={(event) => this.imageClickHandler(event)}
        />
      </Wrapper>,
      document.getElementById('modal-root'),
    );
  }
}

export default ExpandedImage;
