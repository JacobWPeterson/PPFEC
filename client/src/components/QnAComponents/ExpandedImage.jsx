import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
  position: fixed;
  display: table;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 5;
  max-width: 70vw;
  max-height: 60vh;
  overflow: hidden;
  box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);
`;

const Expanded = styled.img`
  display: block;
  max-height: 60vh;
`;

const ExpandedImage = (props) => ReactDOM.createPortal(
  <Wrapper>
    <Expanded
      key={props.id}
      src={props.src}
      alt={props.alt}
      onMouseDown={props.func}
    />
  </Wrapper>,
  document.getElementById('modal-root') || document.createElement('div'), // for testing purposes,
);

export default ExpandedImage;
