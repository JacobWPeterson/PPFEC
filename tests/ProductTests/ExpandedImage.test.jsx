/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';
import styled from 'styled-components';

import ExpandedImage from '../../client/src/components/ProductSubs/ExpandedImage.jsx';

const props = {
  source: 'https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80',
  images: [{ thumbnail_url: 'https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80' }, { thumbnail_url: 'https://images.unsplash.com/photo-1518687338977-a84d3086a934?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1518687338977-a84d3086a934?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=652&q=80' }, { thumbnail_url: 'https://images.unsplash.com/photo-1556812191-381c7e7d96d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1556812191-381c7e7d96d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2982&q=80' }, { thumbnail_url: 'https://images.unsplash.com/photo-1526330563440-3ae2174b6bce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1526330563440-3ae2174b6bce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1655&q=80' }, { thumbnail_url: 'https://images.unsplash.com/photo-1516199707327-5399434d0aa4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1516199707327-5399434d0aa4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1083&q=80' }, { thumbnail_url: 'https://images.unsplash.com/photo-1541444446610-749d3299b35e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1541444446610-749d3299b35e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80' }],
};

describe('ExpandedImage', () => {
  const ExpandedImageComponent = mount(<ExpandedImage {...props} debug />);

  it('Should render correctly', () => {
    expect(ExpandedImageComponent).toMatchSnapshot();
  });
});
