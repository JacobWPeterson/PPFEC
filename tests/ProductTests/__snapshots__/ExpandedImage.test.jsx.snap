// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandedImage Should render correctly 1`] = `
<ExpandedImage
  debug={true}
  images={
    Array [
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80",
      },
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1518687338977-a84d3086a934?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1518687338977-a84d3086a934?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=652&q=80",
      },
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1556812191-381c7e7d96d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1556812191-381c7e7d96d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2982&q=80",
      },
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1526330563440-3ae2174b6bce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1526330563440-3ae2174b6bce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1655&q=80",
      },
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1516199707327-5399434d0aa4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1516199707327-5399434d0aa4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1083&q=80",
      },
      Object {
        "thumbnail_url": "https://images.unsplash.com/photo-1541444446610-749d3299b35e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
        "url": "https://images.unsplash.com/photo-1541444446610-749d3299b35e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80",
      },
    ]
  }
  source="https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80"
>
  <Portal
    containerInfo={
      <div>
        <div
          class="ExpandedImage__Wrapper-sc-1g7vqin-0 iJrbzo"
        >
          <div
            class="ExpandedImage__ModalLeftArrow-sc-1g7vqin-2 ZdNkZ"
          >
            <span
              aria-hidden="true"
              class="fa fa-angle-left fa-2x"
              id="-1"
            />
          </div>
          <img
            class="ExpandedImage__Expanded-sc-1g7vqin-1 iLUdGR"
            x="0"
            y="0"
          />
          <div
            class="ExpandedImage__ModalRightArrow-sc-1g7vqin-3 kMeIWk"
          >
            <span
              aria-hidden="true"
              class="fa fa-angle-right fa-2x"
              id="1"
            />
          </div>
          <div
            class="ExpandedImage__IconHolder-sc-1g7vqin-4 iGQpBV"
          >
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="0"
              type="button"
            />
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="1"
              type="button"
            />
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="2"
              type="button"
            />
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="3"
              type="button"
            />
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="4"
              type="button"
            />
            <div
              class="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
              id="5"
              type="button"
            />
          </div>
        </div>
      </div>
    }
  >
    <ExpandedImage__Wrapper>
      <div
        className="ExpandedImage__Wrapper-sc-1g7vqin-0 iJrbzo"
      >
        <ExpandedImage__ModalLeftArrow
          onMouseDown={[Function]}
        >
          <div
            className="ExpandedImage__ModalLeftArrow-sc-1g7vqin-2 ZdNkZ"
            onMouseDown={[Function]}
          >
            <FontAwesome
              id="-1"
              name="angle-left"
              size="2x"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-left fa-2x"
                id="-1"
              />
            </FontAwesome>
          </div>
        </ExpandedImage__ModalLeftArrow>
        <ExpandedImage__Expanded
          onMouseDown={[Function]}
          x={0}
          y={0}
          zoomed={false}
        >
          <img
            className="ExpandedImage__Expanded-sc-1g7vqin-1 iLUdGR"
            onMouseDown={[Function]}
            x={0}
            y={0}
          />
        </ExpandedImage__Expanded>
        <ExpandedImage__ModalRightArrow
          onMouseDown={[Function]}
        >
          <div
            className="ExpandedImage__ModalRightArrow-sc-1g7vqin-3 kMeIWk"
            onMouseDown={[Function]}
          >
            <FontAwesome
              id="1"
              name="angle-right"
              size="2x"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-right fa-2x"
                id="1"
              />
            </FontAwesome>
          </div>
        </ExpandedImage__ModalRightArrow>
        <ExpandedImage__IconHolder
          onMouseDown={[Function]}
        >
          <div
            className="ExpandedImage__IconHolder-sc-1g7vqin-4 iGQpBV"
            onMouseDown={[Function]}
          >
            <ExpandedImage__Icon
              id={0}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={0}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
            <ExpandedImage__Icon
              id={1}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={1}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
            <ExpandedImage__Icon
              id={2}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={2}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
            <ExpandedImage__Icon
              id={3}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={3}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
            <ExpandedImage__Icon
              id={4}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={4}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
            <ExpandedImage__Icon
              id={5}
              key="NaN"
              onMouseDown={[Function]}
              type="button"
            >
              <div
                className="ExpandedImage__Icon-sc-1g7vqin-6 hFSqpB"
                id={5}
                onMouseDown={[Function]}
                type="button"
              />
            </ExpandedImage__Icon>
          </div>
        </ExpandedImage__IconHolder>
      </div>
    </ExpandedImage__Wrapper>
  </Portal>
</ExpandedImage>
`;
